tests/webdriver/color/color-common.js

```diff
@@ -0,0 +1,40 @@
+function generateScript(renderMethod) {
+  var testScript = function() {
+    var box = Crafty.e('2D, $renderMethod')
+                    .attr({ x: 0, y: 0, w: 320, h: 240 });
+
+    function initial() {
+      box.addComponent('Color');
+      signalBarrier('initial');
+    }
+
+    function opaque() {
+      box.color('rgb(0, 255, 0)');
+      signalBarrier('opaque');
+    }
+
+    function transparent() {
+      box.color('rgba(0, 0, 255, 0.5)');
+      signalBarrier('transparent');
+    }
+
+    waitBarrier('initial', initial);
+    waitBarrier('opaque', opaque);
+    waitBarrier('transparent', transparent);
+  };
+
+  return testScript.toString().replace('$renderMethod', renderMethod);
```

Comment:
Yeah, the problem is the whole function must not have any external dependencies, as it will be serialized as-is into a test html page.   
Alternatively, you could introduce a `renderMethod` parameter, then you would have to wrap the function into an anonymous function call that also supplies the `renderMethod` parameter. I just thought this was more readable, as you have to manually prepend and append the necessary parentheses for that to work.